src/bundle.c

```diff
@@ -171,30 +175,236 @@ static int unload_tracked_bundle(const char *bundle_name, struct list **subs)
+/* Return recursive list of included bundles */
+int show_included_bundles(char *bundle_name)
+{
+	int ret = 0;
+	int current_version = CURRENT_OS_VERSION;
+	struct list *subs = NULL;
+	struct list *deps = NULL;
+	struct manifest *mom = NULL;
+
+	current_version = get_current_version(path_prefix);
+	if (current_version < 0) {
+		fprintf(stderr, "Error: Unable to determine current OS version\n");
+		ret = ECURRENT_VERSION;
+		goto out;
+	}
+
+	mom = load_mom(current_version, false);
+	if (!mom) {
+		fprintf(stderr, "Cannot load official manifest MoM for version %i\n", current_version);
+		ret = EMOM_NOTFOUND;
+		goto out;
+	}
+
+	// add_subscriptions takes a list, so construct one with only bundle_name
+	struct list *bundles = NULL;
+	bundles = list_prepend_data(bundles, bundle_name);
+	ret = add_subscriptions(bundles, &subs, current_version, mom, true, 0);
+	list_free_list(bundles);
+	if (ret != add_sub_NEW) {
+		// something went wrong or there were no includes, print a message and exit
+		char *m = NULL;
+		if (ret & add_sub_ERR) {
+			string_or_die(&m, "Processing error");
+		} else if (ret & add_sub_BADNAME) {
+			string_or_die(&m, "Bad bundle name detected");
+		} else {
+			string_or_die(&m, "Unknown error");
+		}
+
+		fprintf(stderr, "Error: %s - Aborting\n", m);
+		free(m);
+		ret = 1;
+		goto out;
+	}
+	deps = recurse_manifest(mom, subs, NULL, false);
+	if (!deps) {
+		fprintf(stderr, "Error: Cannot load included bundles\n");
+		ret = ERECURSE_MANIFEST;
+		goto out;
+	}
+
+	/* deps now includes the bundle indicated by bundle_name
+	 * if deps only has one bundle in it, no included packages were found */
+	if (list_len(deps) == 1) {
+		fprintf(stderr, "No included bundles\n");
+		ret = 0;
+		goto out;
+	}
+
+	fprintf(stderr, "Bundles included by %s:\n\n", bundle_name);
+
+	struct list *iter;
+	iter = list_head(deps);
+	while (iter) {
+		struct manifest *included_bundle = iter->data;
+		iter = iter->next;
+		// deps includes the bundle_name bundle, skip it
+		if (strcmp(bundle_name, included_bundle->component) == 0) {
+			continue;
+		}
+
+		printf("%s\n", included_bundle->component);
+	}
+
+	ret = 0;
+
+out:
+	if (mom) {
+		free_manifest(mom);
+	}
+
+	if (deps) {
+		list_free_list_and_data(deps, free_manifest_data);
+	}
+
+	if (subs) {
+		free_subscriptions(&subs);
+	}
+
+	return ret;
+}
+
+int show_bundle_reqd_by(const char *bundle_name, bool server)
+{
+	int ret = 0;
+	int version = CURRENT_OS_VERSION;
+	struct manifest *current_manifest = NULL;
+	struct list *subs = NULL;
+	struct list *reqd_by = NULL;
+
+	if (!server && !is_tracked_bundle(bundle_name)) {
+		fprintf(stderr, "Error: Bundle \"%s\" does not seem to be installed\n", bundle_name);
+		fprintf(stderr, "       try passing --all to check uninstalled bundles\n");
+		ret = EBUNDLE_NOT_TRACKED;
+		goto out;
+	}
+
+	version = server ? get_latest_version() : get_current_version(path_prefix);
+	if (version < 0) {
+		fprintf(stderr, "Error: Unable to determine current OS version\n");
```

Comment:
There doesn't seem to be any issue if version should be the current version since get_current_version returns -1 it can't determine the current version.

Reply:
I was referring to the case when `get_latest_version()` is called, but you have amended this line. Ignore the comment :)